### PR TITLE
Delete old courses 

### PR DESCRIPTION
**Pull request by hankewyczz**
# Purpose

###### A brief description of the purpose of the PR's changes for the project.

<br> 

Actually implement [this ticket ](https://trello.com/c/NrnPz8Om/309-old-courses-not-being-deleted)(and add tests to make sure it works lol)

# Tickets

###### A link to any tickets this PR is associated with on Trello.

- https://trello.com/c/NrnPz8Om/309-old-courses-not-being-deleted

# Contributors

###### Anyone who contributed to this PR for future reference.

- @hankewyczz 

# Notes

###### A list of miscellaneous notes for this pull request, such as whether we need to add something later for production, or that it was time boxed, etc.

<br>

This would fail on updates, but worked on scrapes. The reason being - we only took into account the termIds of the classes, not the sections. Since the updater doesn't consider classes (only sections), no old sections were being deleted. 

# Checklist

- [x] Filled out PR template :wink:
- [x] Approved by designers
- [ ] Is passing linting checks
- [ ] Is passing tsc
- [ ] Is passing existing tests
- [ ] Has documentation and comments in code
- [ ] Has test coverage for code
- [ ] Will have clear squash commit message

<br>
@sandboxnu/searchneu

### CODE REVIEW

**Comment by hankewyczz** [comment]
> [Coverage Status](https://coveralls.io/builds/47950584)
> 
> Coverage decreased (-24.02%) to 30.832% when pulling **[b957709](https://github.com/sandboxnu/course-catalog-api/commit/b9577097b960ba5455f17fb236eb8ce96da03a18) on zh/fix-outdated-section-deletion** into **[86dc41c](https://github.com/sandboxnu/course-catalog-api/commit/86dc41c2d36adeec89b276214bea625059d231e7) on master**.

fake news

File: tests/workflow_tests/updater.test.git.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+              sections {
+                termId
+              }
+            }
+          }
+        }
+      }
+    `);
+
+    expect(res.data?.search.nodes[0].sections.length).toBe(3);
```
Comment (andrewydai) [inline tests/workflow_tests/updater.test.git.ts:29]: kinda small but i'd like this test to specifically check that that section isn't there. Just from following the code change it looks like 2 sections are inserted then 1 is removed via updater. As a result testing that 3 sections exist doesn't seem immediately clear to me as a test of 1900's class being not there.

**Review by andrewydai — CHANGES_REQUESTED** [review]
🏎️  i'm just doing random emojis but left a comment

File: services/dumpProcessor.ts [hunk 2]
```diff
@@ -338,12 +338,7 @@ class DumpProcessor {
     ]) as Prisma.ProfessorCreateInput;
   }
 
-  processCourse(
-    classInfo: any,
-    coveredTerms: Set<string> = new Set()
-  ): Prisma.CourseCreateInput {
-    coveredTerms.add(classInfo.termId);
-
+  processCourse(classInfo: any): Prisma.CourseCreateInput {
```
Comment (megandouli) [inline services/dumpProcessor.ts:341]: it looks like `processCourse` isn't being used for anything - can we delete?

File: services/updater.ts [hunk 3]
```diff
@@ -60,14 +60,17 @@ class Updater {
...
-    setInterval(async () => {
-      // TEMP / TODO - REMOVE / DO NOT LEAVE HERE PLEASE
-      // Temp fix to address Prisma connection pool issues
-      // https://github.com/prisma/prisma/issues/7249#issuecomment-1059719644
-      await this.updateOrExit();
-      await prisma.$disconnect();
-      macros.log("Disconnected Prisma");
-    }, intervalTime);
+    // Flag only used for testing, since we only need the updater to run once
+    if (!process.env.UPDATE_ONLY_ONCE) {
```
Comment (megandouli) [inline services/updater.ts:64]: instead of a new flag here, can we just check `macros.test`?

File: services/dumpProcessor.ts (hunk 2 above)
Comment (hankewyczz) [reply services/dumpProcessor.ts:341]: Moving it into the tests - we use it to prepare course objects for a few tests

File: services/updater.ts (hunk 3 above)
Comment (hankewyczz) [reply services/updater.ts:64]: Parts of the workflow functionality depend on the node_env being `dev`, so changing to test isn't ideal.

**Review by megandouli — APPROVED** [review]

**Pull request merged**